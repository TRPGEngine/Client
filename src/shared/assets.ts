import failedImg from '@web/assets/img/img_fail.png';
import robotImg from '@web/assets/img/robot_dark.svg';

/**
 * 获取默认图片
 */
export function getDefaultImage({ fileUrl }: { fileUrl: string }) {
  return {
    user: '/src/assets/img/gugugu1.png',
    /**
     * @deprecated 使用Avatar组件生成默认头像
     */
    getUser(name: string) {
      if (name) {
        return `${fileUrl}/file/avatar/svg?name=${name}`;
      } else {
        return 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'; // 一像素透明图片
      }
    },
    group: '/src/web/assets/img/gugugu1.png',
    /**
     * @deprecated 使用Avatar组件生成默认头像
     */
    getGroup(name: string) {
      // 同getUser
      if (name) {
        return `${fileUrl}/file/avatar/svg?name=${name}`;
      } else {
        return 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'; // 一像素透明图片
      }
    },
    trpgsystem: '/src/web/assets/img/system_notice.png',
    robot: robotImg,
    actor: '',
    chatimg_fail: failedImg,
    file: {
      default: '/src/web/assets/img/file/default.png',
      pdf: '/src/web/assets/img/file/pdf.png',
      excel: '/src/web/assets/img/file/excel.png',
      ppt: '/src/web/assets/img/file/ppt.png',
      word: '/src/web/assets/img/file/word.png',
      txt: '/src/web/assets/img/file/txt.png',
      pic: '/src/web/assets/img/file/pic.png',
    },
  };
}
