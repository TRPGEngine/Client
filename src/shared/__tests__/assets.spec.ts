import { getDefaultImage } from '@shared/assets';

describe('getDefaultImage', () => {
  const defaultImage = getDefaultImage({ fileUrl: '[fileUrl]' });

  test('snapshot', () => {
    expect(defaultImage).toMatchSnapshot();
  });
});
