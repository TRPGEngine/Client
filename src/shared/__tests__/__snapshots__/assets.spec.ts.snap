// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDefaultImage snapshot 1`] = `
Object {
  "actor": "",
  "chatimg_fail": "img_fail.png",
  "file": Object {
    "default": "default.png",
    "excel": "excel.png",
    "pdf": "pdf.png",
    "pic": "pic.png",
    "ppt": "ppt.png",
    "txt": "txt.png",
    "word": "word.png",
  },
  "getGroup": [Function],
  "getUser": [Function],
  "group": "gugugu1.png",
  "robot": "robot_dark.svg",
  "trpgsystem": "system_notice.png",
  "user": "gugugu1.png",
}
`;
